engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -282,23 +271,34 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     *
     * @return a RoundRobinPool of actors based on this.factory.
     */
-  private def fileContentsHasherActor: ActorRef = {
+  private[execution] lazy val fileContentsHasherActor: ActorRef = {
     factorySingletonRouters.getOrElseUpdate(factory,
       context.actorOf(
         RoundRobinPool(factory.fileHashingWorkerCount).props(factory.fileContentsHasherActorProps),
         "FileContentsHasherActor")
     )
   }
 
+  def makeFetchCachedResultsActor(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput]): Unit = context.actorOf(FetchCachedResultsActor.props(cacheHit, taskOutputs, self, new CallCache(SingletonServicesStore.databaseInterface)))
   def fetchCachedResults(data: ResponsePendingData, taskOutputs: Seq[TaskOutput], cacheHit: CacheHit) = {
-    context.actorOf(FetchCachedResultsActor.props(cacheHit, taskOutputs, self, new CallCache(SingletonServicesStore.databaseInterface)))
+    makeFetchCachedResultsActor(cacheHit, taskOutputs)
     goto(FetchingCachedOutputsFromDatabase)
   }
 
   def makeBackendCopyCacheHit(cacheHit: CacheHit, cachedJobOutputs: JobOutputs, data: ResponsePendingData) = {
-    val cacheHitCopyActor = context.actorOf(callCachingActorProps, buildCacheHitCopyingActorName(data.jobDescriptor))
-    cacheHitCopyActor ! CopyOutputsCommand(cachedJobOutputs)
-    goto(BackendIsCopyingCachedOutputs)
+    factory.cacheHitCopyingActorProps match {
+      case Some(propsMaker) =>
+        val backendCacheHitCopyingActorProps = propsMaker(data.jobDescriptor, initializationData, serviceRegistryActor)
+        val cacheHitCopyActor = context.actorOf(backendCacheHitCopyingActorProps, buildCacheHitCopyingActorName(data.jobDescriptor))
+        cacheHitCopyActor ! CopyOutputsCommand(cachedJobOutputs)
+        goto(BackendIsCopyingCachedOutputs)
+      case None =>
+        // This should be impossible with the FSM, but luckily, we CAN recover if this happens
+        val errorMessage = "Call caching copying impossible with no copy actor props! (Programmer error!)"
```

Comment:
The wording might be more clear: if `factory.cacheHitCopyingActorProps` is empty call cache reading should have been turned off and it shouldn't be possible to get here